docs/reference/query-rules/apis/delete-query-ruleset.asciidoc

```diff
@@ -0,0 +1,48 @@
+[role="xpack"]
+[[delete-query-ruleset]]
+=== Delete query ruleset
+
+preview::[]
+
+++++
+<titleabbrev>Delete query ruleset</titleabbrev>
+++++
+
+Removes a query ruleset and its associated data.
+This is a destructive action that is not recoverable.
+
+[[delete-query-ruleset-request]]
+==== {api-request-title}
+
+`DELETE _query_rules/<ruleset_id>`
+
+[[delete-query-ruleset-prereq]]
+==== {api-prereq-title}
+
+Requires the `manage_cluster` privilege.
+
+[[delete-query_ruleset-path-params]]
+==== {api-path-parms-title}
+
+`<ruleset_id>`::
+(Required, string)
+
+[[delete-query-ruleset-response-codes]]
+==== {api-response-codes-title}
+
+`400`::
+The `ruleset_id` was not provided.
+
+`404` (Missing resources)::
+No query ruleset matching `ruleset_id` could be found.
+
+[[delete-query-ruleset-example]]
+==== {api-examples-title}
+
+The following example deletes the query ruleset named `my-ruleset`:
+
+[source,console]
+----
+DELETE _query_rules/my-ruleset/
+----
+// TEST[skip:TBD]
```

Comment:
It would be nice to add tests once https://github.com/elastic/elasticsearch/pull/97250 is merged. Let me know if you need a hand with those.

Reply:
Thanks @abdonpijpelink - Are there docs that we can get started with here? I noticed that our search application docs also don't have tests.

Reply:
There are basic docs about snippet testing here: https://github.com/elastic/elasticsearch/tree/main/docs#test-code-snippets. You could use the index template docs as an example: [put](https://github.com/elastic/elasticsearch/blob/main/docs/reference/indices/put-index-template.asciidoc?plain=1)/[get](https://github.com/elastic/elasticsearch/blob/main/docs/reference/indices/get-index-template.asciidoc?plain=1)/[delete](https://github.com/elastic/elasticsearch/blob/main/docs/reference/indices/delete-index-template.asciidoc?plain=1)

Reply:
Thanks, added tests!